app/gameengine.py

```diff
@@ -2711,6 +2785,20 @@ def is_condition_met(self, effect_player: PlayerState, source_card_id, condition
                 return False
             case Condition.Condition_PlayedSupportThisTurn:
                 return effect_player.played_support_this_turn
+            case Condition.Condition_RevealedCardsHasSameType:
+                revealed_cards = effect_player.last_revealed_cards
+                if len(revealed_cards) == 0:
+                    return False
+                match condition.get("condition_same_type"):
+                    case "holomem_same_bloom":
+                        # Cards should be holomem and of the same bloom level (Debut is level 0)
+                        base_card = revealed_cards[0] # the card that the rest of the cards will be compared to
+                        if not is_card_holomem(base_card):
+                            return False
+                        card_type = base_card["card_type"]
+                        bloom_level = base_card.get("bloom_level", 0)
+                        return all([card["card_type"] == card_type and card.get("bloom_level", 0) == bloom_level for card in revealed_cards[1:]])
```

Comment:
I feel like a function are_cards_matching_bloom_level(card1, card2) might be useful down the line?
or i guess an array of cards as a single param